### ISSUE TITLE
Fix the functionality of 'Skip to Main Content' link

### ISSUE THREAD

**Issue by ltsheu** [18F/dol-whd-14c#307]
Notes from L. Sheu:

**Branch:** lsheu-a11y-skip-link

In this file:
DOL.WHD.Section14c.Web/src/v4/dol-header.component.ts

I have added a new "Skip to Main Content" link just above the DOL header:

`<a href="#mainContent" class="dol-skip-to-main">skip to main content</a>`

The link appears on the page when it receives focus (as from keyboard tabbing). See Note Below.

**Issue:** This is an anchor link that targets `<main id="mainContent">`. However, because this is an Angular site, the link is not behaving as expected. **Currently the link causes the url to reset to /#!/#mainContent, which displays the landing page. We need the link to go to #mainContent within the form page that the user is on.** (For example, if they are on the Application Info page, the "Skip to Main Content" link should go to`<main id="mainContent">` on the Application Info page.) 

**NOTE:**  To see the "Skip To Main Content" link appear in the page, start your cursor in the browser's address bar, then tab once. The link will receive focus and appear in the top left corner of the page. Once clicked (losing focus), the link will disappear.

**Comment by binwang89** [comment]
@davoodharun @ltsheu I just tested the skip main content function for Application Info page, However, the skip function does not work properly.  

below is the steps that I have done for testing:
1. Refresh (F5) the page and test the skip link's functionality. 
2. Tab to the skip function.
 The skip  function must be keyboard accessible and receive a  visual focus. 
3. Select the function with  Enter.
4. Tab again. This step is  necessary to determine whether the skip link functions properly.
5. The focus should move beyond the identified repetitive content to an interactive  element.

**Comment by ltsheu** [comment]
@binwang89 I believe the part where it skips over repetitive content happens at step 3, when the user hits Enter. Step 3 brings the user to the main content region (in the process, it skips over the header and top navigation). I am not sure step 4 and 5 are necessary for testing this? At step 3, you should see the page visually scroll to the main content region of the page.

**Comment by binwang89** [comment]
@ltsheu @davoodharun  Yes, the page vusually scroll down, but when user use the tab key again, it is going to the Dashboard screen instead of staying within the same screen.  I put this user story back to "In Progress"

**Comment by davoodharun** [comment]
@binwang89 I'm not sure what you mean. If I am understanding you correctly, you start on a page (i.e. sections/assurances), click the address bar, hit tab, see the button appear, click the button, the page scrolls down. You are saying when you hit the tab key again it goes to the Dashboard?

If so, that wasn't captured in the user story. The user story only says that the page should scroll to #mainContent when the button is clicked.

**Comment by jmmcnj** [comment]
Tests were written for directive but tests not written for karma yet, issues with karma test actions, need to investigate issue and see path going forward, huddle around

**Comment by binwang89** [comment]
@ltsheu @davoodharun @jmmcnj I just tested the skip link this morning, it works fine. i am going to move this story to next pipeline.

**Comment by phirefly** [comment]
@ltsheu We are doing QA on this but can't seem to find what page this is on. Can you post a link to this page? Thank you!

**Comment by ltsheu** [comment]
@phirefly Go to any page, for example https://dol-whd-section14c-int.aisdemos.com/#!/section/app-info. With your cursor in the address bar of your browser, hit the tab key until you see a "Skip to Main Content" link appear in the top left corner of the page. While that link is visible, you should be able to use the Enter key or click on the link to skip to the main content of the page.

**Comment by davoodharun** [comment]
@phirefly Just in case, make sure you have your dev tools in your browser closed. It will not work with that open.
